src/highdicom/sr/templates.py

```diff
@@ -1935,20 +2888,515 @@ def __init__(
+
+        Parameters
+        ----------
+        tracking_uid: str, optional
+            Unique tracking identifier
+        finding_type: Union[highdicom.sr.coding.CodedConcept, pydicom.sr.coding.Code], optional
+            Finding
+        finding_site: Union[highdicom.sr.coding.CodedConcept, pydicom.sr.coding.Code], optional
+            Finding site
+        graphic_type: highdicom.sr.enum.GraphicTypeValues3D, optional
+            Graphic type of image region
+
+        Returns
+        -------
+        List[highdicom.sr.templates.VolumetricROIMeasurementsAndQualitativeEvaluations]
+            Sequence of content items for each matched measurement group
+
+        """  # noqa
+        sequences = []
+        measurement_group_items = self._find_measurement_groups()
+        for group_item in measurement_group_items:
+            if group_item.template_id is not None:
+                if group_item.template_id != '1411':
+                    continue
+            else:
+                contains_rois = _contains_volumetric_rois(group_item)
+                if not(contains_rois):
+                    continue
+
+            matches = []
+            if finding_type is not None:
+                matches_finding = _contains_code_items(
+                    group_item,
+                    name=codes.DCM.Finding,
+                    value=finding_type
+                )
+                matches.append(matches_finding)
+            if finding_site is not None:
+                matches_finding_sites = _contains_code_items(
+                    group_item,
+                    name=codes.SCT.FindingSite,
+                    value=finding_site
+                )
+                matches.append(matches_finding_sites)
+            if tracking_uid is not None:
+                matches_tracking_uid = _contains_text_items(
+                    group_item,
+                    name=codes.DCM.TrackingUniqueIdentifier,
+                    value=tracking_uid
+                )
+                matches.append(matches_tracking_uid)
+            if graphic_type is not None:
+                matches_graphic_type = _contains_scoord_items(
+                    group_item,
+                    name=codes.DCM.ImageRegion,
+                    graphic_type=graphic_type
+                )
+                matches_graphic_type |= _contains_scoord3d_items(
+                    group_item,
+                    name=codes.DCM.VolumeSurface,
+                    graphic_type=graphic_type
+                )
+                matches.append(matches_graphic_type)
+
+            seq = VolumetricROIMeasurementsAndQualitativeEvaluations.from_sequence(  # noqa
+                [group_item]
+            )
+            if len(matches) == 0:
+                sequences.append(seq)
+            else:
+                if any(matches):
+                    sequences.append(seq)
+
+        return sequences
+
+    def get_image_measurments(
```

Comment:
Good idea!

Reply:
We may want to consider this for the other methods as well. However, for the planar/volumetric ROI methods, it would only work for the segmentation source images

Reply:
Yes, agree for planar ROIs. Not sure whether it makes sense for volumetric ROIs, which can straddle multiple frames? Could be convinced either way

Reply:
What did we intend`reference_type` to be again? The name of the content item (e.g., `pydicom.sr.codes.DCM.ImageRegion`)?

Reply:
I think so. Or I think you were going to ponder having an enum whose values are codes

Reply:
Take a look at 36a3f709e444b9eb1735de9b5a990b92ba943d95. What do you think?

Reply:
Added unit tests for the planar ROI queries via c9fd35b5057fdcff6afd9c2599470ad1ec10c092